azure-pipelines.yml

```diff
@@ -59,7 +59,9 @@ jobs:
       artifactName: test-signing
       targetPath: $(Build.StagingDirectory)/test-signing
   - bash: |
+      ls -l '$(Build.StagingDirectory)'\\test-signing\\installer.exe
       "/C/Program Files/dotnet/dotnet.exe" tool install --global AzureSignTool
+      ls -l /C/Users/u/.dotnet/tools/azuresigntool
```

Comment:
looking at this I wonder if we might be screwing yourself over with path conversion somewhere. I vaguely recall that bash’s automatic path conversion caused very weird issues in the past in a few places and the easiest solution was to just use powershell for those tasks.

Reply:
There is a bad interaction between Bash escaping and YAML escaping: lines like
```bash
$(Build.StagingDirectory)/test.exe
```
will get mangled because Azure YAML substitution goes first, so Bash sees
```bash
C:\Path\With\Backslashes/test.exe
```
and it interprets the backslashes as escaping characters, yielding
```bash
C:PathWithBackslashes/test.exe
```
which does not exist. Adding single quotes around the YAML substitution makes it look a bit weird from a Bash perspective (single quotes meaning no substitution) but make this work properly as the YAML substitution ignores the single quotes, but the single quotes do tell Bash to not interpret the backslashes as escape characters.

If you were thinking of another issue, then I'm not (yet) aware of it.